src/tools/compiletest/src/common.rs

```diff
@@ -265,13 +265,23 @@ pub struct Config {
     /// cf. [`Self::remote_test_client`] and [`Self::runner`].
     pub target_run_lib_path: Utf8PathBuf,
 
-    /// Path to the *staged*  `rustc`-under-test. Unless forced, this `rustc` is *staged*, and must
-    /// not be confused with [`Self::stage0_rustc_path`].
+    /// Path to the *staged* `rustc`-under-test for non-`ui-fulldeps` test suites. Path to stage 0
+    /// `rustc` for the `ui-fulldeps` test suite.
+    ///
+    /// For `ui-fulldeps`, see also [`Self::query_rustc_path`] for the *staged* `rustc` that is used
+    /// to query target information from.
     ///
     /// For example:
     /// - `/home/ferris/rust/build/x86_64-unknown-linux-gnu/stage1/bin/rustc`
     ///
-    /// FIXME: maybe rename this to reflect that this is the `rustc`-under-test.
+    /// # Note on forced stage0
+    ///
+    /// It is possible for this `rustc` to be a stage 0 `rustc` if explicitly configured with the
+    /// bootstrap option `build.compiletest-allow-stage0=true` and specifying `--stage=0`.
+    ///
+    /// FIXME(jieyouxu): fixup `ui-fulldeps` logic to consistently use `rustc_path` as (always) a
+    /// staged in-tree `rustc`, and use `stage0_rustc_path` where previously `ui-fulldeps` used
+    /// `rustc_path` for passing the stage 0 `rustc.`
```

Comment:
Future work: Hmm, I guess we could try moving the special-case logic into compiletest itself.

So if compiletest would try to compile something with the stage1 compiler in ui-fulldeps, it automatically uses the stage0 compiler instead.

Reply:
Now that I think about it, this FIXME isn't quite right for `ui-fulldeps` at stage 2, so I guess the current setup is fine.